components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -681,8 +700,11 @@ public static String getUserConsentURL(OAuth2Parameters params, String loggedInU
                 consentPage += "&tenantDomain=" + getSPTenantDomainFromClientId(params.getClientId());
 
                 consentPage = consentPage + "&" + OAuthConstants.OAuth20Params.SCOPE + "=" + URLEncoder.encode
-                        (getFilteredScopes(params), UTF_8) + "&" + OAuthConstants.SESSION_DATA_KEY_CONSENT
-                        + "=" + URLEncoder.encode(sessionDataKeyConsent, UTF_8) + "&spQueryParams=" + queryString;
+                        (getFilteredScopes(params, user), UTF_8) + "&" + OAuthConstants.SESSION_DATA_KEY_CONSENT
```

Comment:
Any particular reason why we need to send both scope and consent required scopes seperatly as query params? cant we only send the consent required scopes as the scopes?

Reply:
No. Scope parameter is used when building the token. That's why a new parameter was introduced to track consent required scopes.

Reply:
Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1548/commits/2860df2de9cbb8dfe62da23204474dd122a25de7